### ISSUE TITLE
All previously configured channels, stream providers, and settings disappeared after updating to the latest docker image

### ISSUE THREAD

**Issue by successtheman** [cabernetwork/cabernet#134]
All my settings that I had configured are gone along with the stream providers etc... this appears to have happened right after updating to the latest docker image (as of this issue's creation) from January 29th 

Previously I had manually updated with the web ui but I think maybe the latest changes with the docker image preventing updates within the web ui might have something to do with it.

I still have backups of the configurations in the backups directory, so is there a way to fix this?

**Comment by successtheman** [comment]
> Can you provide some settings that you changed previously?
> 
> Did you check the new instructions for docker? https://github.com/cabernetwork/cabernet?tab=readme-ov-file#4-docker
> 
> You no longer add `app/` to volume, instead only data and `plugins_ext`
> 
> > Previously I had manually updated with the web ui but I think maybe the latest changes with the docker image preventing updates within the web ui might have something to do with it.
> 
> That causes issues with docker and shouldn't be done, that's why is blocked now.

I am going to try switching the mounted directories to just `data` and `plugins_ext` and see if that helps. I also did read over the instructions but it's a bit confusing as it doesn't really indicate what is new, so that's why I asked here. I will report back with the result.

**Comment by successtheman** [comment]
> copy your previous files from `data/` to new `-v data` path of simply use the same data directory as volume and should work with your previous settings, do the same do `plugins_ext`
> 
> > a bit confusing as it doesn't really indicate what is new
> 
> Check the PR #124
> 
> PS: Also there's a new pull request with more options for docker, where you can set some settings for the container for easier as faster deployment, not merged yet, maybe in the future.

Mounting `data/` and `plugins_ext/` instead of `app/` seems to do the trick and it is working once again. 

Is there anything I should be doing regarding `app/secret/` because I never had that file to begin with and don't know how to get it if it is needed.

**Comment by successtheman** [comment]
> >Is there anything I should be doing regarding app/secret/ because I never had that file to begin with and don't know how to get it if it is needed.
> 
> Only if you enable encryption 

Is it necessary for future updates to work, and is there a downside to not enabling it? I usually just take a tar backup of the entire cabernet directory and upload somewhere.

Additionally the web UI itself is not currently password protected so until that happens I really don't see the point to enable encryption. Web UI access being password protected would honestly be a welcomed addition to the software (it can be achieved via nginx tunneling I believe externally but it would be nice to have it natively in cabernet).

**Comment by successtheman** [comment]
> >Is there anything I should be doing regarding app/secret/ because I never had that file to begin with and don't know how to get it if it is needed.
> 
> Only if you enable encryption 

> >Is there anything I should be doing regarding app/secret/ because I never had that file to begin with and don't know how to get it if it is needed.
> 
> Only if you enable encryption 

Is it necessary for future updates to work, and is there a downside to not enabling it? I usually just take a tar backup of the entire cabernet directory and upload somewhere.

Additionally the web UI itself is not currently password protected so until that happens I really don't see the point to enable encryption. Web UI access being password protected would honestly be a welcomed addition to the software (it can be achieved via nginx tunneling I believe externally but it would be nice to have it natively in cabernet).

Also thanks for the help in solving my problem, I closed the issue since it has been resolved even though we are still discussing other stuff (I think it will let us keep talking while it's closed).

**Comment by successtheman** [comment]
> Only if you enable encryption

@Generator 

Is encryption necessary for future updates to work, and is there a downside to not enabling it? I usually just take a tar backup of the entire cabernet directory and upload somewhere.

Additionally the web UI itself is not currently password protected so until that happens I really don't see the point to enable encryption. Web UI access being password protected would honestly be a welcomed addition to the software (it can be achieved via nginx tunneling I believe externally but it would be nice to have it natively in cabernet).

Also thanks for the help in solving my problem, I closed the issue since it has been resolved even though we are still discussing other stuff (I think it will let us keep talking while it's closed).
